### PR TITLE
pdfposter: move out of python-modules fix build pin pypdf2

### PR DESCRIPTION
**Pull request by kirillrdy**
###### Description of changes

pdfposter is not compatible with pypdf3, but given its mainly used as application should be safe to overwrite pypdf

here is upstream MR to migrate to pypdf3
https://gitlab.com/pdftools/pdfposter/-/merge_requests/7

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kirillrdy** [comment]
Result of `nixpkgs-review pr 240414` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.pdfposter</li>
    <li>python310Packages.pdfposter.dist</li>
  </ul>
</details>

**Comment by kirillrdy** [comment]
cc @laikq @mweinelt

**Comment by mweinelt** [comment]
We cannot use overrides within the python package set, as that causes conflicts when packages introduce different versions of the same package into the same python environment.

**Comment by kirillrdy** [comment]
> We cannot use overrides within the python package set, as that causes conflicts when packages introduce different versions of the same package into the same python environment.
> 
> While that might seem unlikely, we don't make exceptions in such cases.

yes, I understand.
From first glance it looks like pdfposter is mean to be used as application,  in which case it probably shouldnt be in python-modules. if thats the case then overrides should be fine.

**Comment by mweinelt** [comment]
Yep, looks like an application to me as well.

**Review by FRidh — APPROVED** [review]

**Review by wamserma — CHANGES_REQUESTED** [review]
only a small request, otherwise LGTM

File: pkgs/applications/misc/pdfposter/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{ lib, python3, fetchPypi }:
+let
+  localPython = python3.override {
+    self = localPython;
+    packageOverrides = self: super: {
+      pypdf2 = super.pypdf2.overridePythonAttrs (oldAttrs: rec {
```
[the remark this answers is not in the document]
Comment (kirillrdy) [reply pkgs/applications/misc/pdfposter/default.nix:8]: yep, makes sense, done

**Pull request merged**